nav2_common/nav2_common/launch/node.py

```diff
@@ -131,255 +88,10 @@ def __init__(
-                    "ros_specific_arguments['remaps'][{}]".format(i),
-                    description='remapping {}'.format(i))]
-                i += 1
-        super().__init__(cmd=cmd, **kwargs)
-        self.__package = package
-        self.__node_executable = node_executable
-        self.__node_name = node_name
-        self.__node_namespace = node_namespace
-        self.__parameters = [] if parameters is None else normalized_params
+        super().__init__(**kwargs)
         self.__use_remappings = use_remappings
-        self.__remappings = [] if remappings is None else remappings
-        self.__arguments = arguments
-
-        self.__expanded_node_name = '<node_name_unspecified>'
-        self.__expanded_node_namespace = ''
-        self.__final_node_name = None  # type: Optional[Text]
-        self.__expanded_parameter_files = None  # type: Optional[List[Text]]
-        self.__expanded_remappings = None  # type: Optional[List[Tuple[Text, Text]]]
-
-        self.__substitutions_performed = False
-
-        self.__logger = launch.logging.get_logger(__name__)
-
-    @staticmethod
-    def parse_nested_parameters(params, parser):
-        """Normalize parameters as expected by Node constructor argument."""
-        def get_nested_dictionary_from_nested_key_value_pairs(params):
-            """Convert nested params in a nested dictionary."""
-            param_dict = {}
-            for param in params:
-                name = tuple(parser.parse_substitution(param.get_attr('name')))
-                value = param.get_attr('value', data_type=None, optional=True)
-                nested_params = param.get_attr('param', data_type=List[Entity], optional=True)
-                if value is not None and nested_params:
-                    raise RuntimeError('param and value attributes are mutually exclusive')
-                elif value is not None:
-                    def normalize_scalar_value(value):
-                        if isinstance(value, str):
-                            value = parser.parse_substitution(value)
-                            if len(value) == 1 and isinstance(value[0], TextSubstitution):
-                                value = value[0].text  # python `str` are not converted like yaml
-                        return value
-                    if isinstance(value, list):
-                        value = [normalize_scalar_value(x) for x in value]
-                    else:
-                        value = normalize_scalar_value(value)
-                    param_dict[name] = value
-                elif nested_params:
-                    param_dict.update({
-                        name: get_nested_dictionary_from_nested_key_value_pairs(nested_params)
-                    })
-                else:
-                    raise RuntimeError('either a value attribute or nested params are needed')
-            return param_dict
-
-        normalized_params = []
-        params_without_from = []
```

Comment:
@orduno I wasn't sure what you were doing here with the `params_without_from` and associated code. Can you explain what this was doing? It seems to work without, but I'm not sure if I'm missing a corner case.